sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -366,24 +457,24 @@ function buildRawSqlParameter(parameter: SqlParameter): RawSqlParameter {
     );
   }
 
-  let type: SqlParameterType;
+  let type: TypeMapForRequestSerialization;
```

Comment:
Would it be easier to just create a Map<> or record type and store it as a literal, rather than having this code? Sometimes data-driven is nicer.

Reply:
Agree.

Reply:
Scoping this out of this PR, will do a followup later.

Reply:
Updated to record type.

Reply:
Going with record type

Reply:
NIT: this isn't what I meant but I don't think this is a big deal. Resolving.